newsService/src/main/java/com/kovan/app/service/NewsService.java

```diff
@@ -63,25 +72,51 @@ public NewsDto getTopHeadlines() {
 
         } while (pageSize <= newsDto.getTotalResults());
 
-        LocalDate dateOnly = LocalDate.parse(
-                articles.getFirst().getPublishedAt(),
-                DateTimeFormatter.ISO_DATE_TIME
-        );
+        List<NewsDto.Article> newArticles = new ArrayList<>();
```

Comment:
Do you think this memory creation required at this time itself? If yes, please give me an explanation

Reply:
Memory creation postponed.